### PR TITLE
test: api-middleware

### PR DESCRIPTION
**Pull request by sean-brydon**
Adds tests to api middleware

- VerifyApiKey
- httpMethods
- addRequestId
- Will add rate limit tests after it has been merged  https://github.com/calcom/cal.com/pull/11783

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -7,5 +7,6 @@
   "typescript.preferences.importModuleSpecifier": "non-relative",
   "spellright.language": ["en"],
   "spellright.documentTypes": ["markdown", "typescript", "typescriptreact"],
-  "tailwindCSS.experimental.classRegex": [["cva\\(([^)]*)\\)", "[\"'`]([^\"'`]*).*?[\"'`]"]]
+  "tailwindCSS.experimental.classRegex": [["cva\\(([^)]*)\\)", "[\"'`]([^\"'`]*).*?[\"'`]"]],
+  "vitest.commandLine": ""
```
Comment (sean-brydon) [inline .vscode/settings.json:11]:
```suggestion
```

File: apps/api/test/lib/bookings/_post.test.ts
Comment (sean-brydon) [inline apps/api/test/lib/bookings/_post.test.ts:1]: Fix tests (These test were never running due to the vitest workspace config) 

Commented out so we can still run tests in API dir

File: vitest.workspace.ts [hunk 2]
```diff
@@ -16,7 +16,7 @@ const workspaces = packagedEmbedTestsOnly
         test: {
           include: ["packages/**/*.{test,spec}.{ts,js}", "apps/**/*.{test,spec}.{ts,js}"],
           // TODO: Ignore the api until tests are fixed
-          exclude: ["apps/api/**/*", "**/node_modules/**/*", "packages/embeds/**/*"],
```
Comment (sean-brydon) [inline vitest.workspace.ts:19]: No longer exclude tests

File: apps/api/test/lib/bookings/_post.test.ts [hunk 3]
```diff
@@ -21,7 +22,7 @@ vi.mock("@calcom/lib/server/i18n", () => {
   };
 });
 
-describe("POST /api/bookings", () => {
+describe.skipIf(true)("POST /api/bookings", () => {
```
Comment (sean-brydon) [inline apps/api/test/lib/bookings/_post.test.ts:25]: Skipping as these tests are always failing - they never used to run due to how vitest.workspace.ts was configured

**Review by keithwillcode — APPROVED** [review]
Great work, @sean-brydon. What do you think about adding tests for the `withMiddleware` file as well. This way we can guarantee the middleware we expect to be wired up is there and in the order we expect.

**Pull request merged**